proc_dash/app.py

```diff
@@ -324,66 +336,142 @@ def process_bagel(contents, filename):
     """
     if contents is None:
         return (
+            None,
             None,
             "Upload a CSV file to begin.",
             [],
             no_update,
             no_update,
             no_update,
+            no_update,
         )
     try:
-        data, sessions, upload_error = util.parse_csv_contents(
-            contents=contents, filename=filename
-        )
+        (
+            overview_df,
+            sessions,
+            pipelines_dict,
+            upload_error,
+        ) = util.parse_csv_contents(contents=contents, filename=filename)
     except Exception as exc:
         print(exc)  # for debugging
         upload_error = "Something went wrong while processing this file."
 
     if upload_error is not None:
         return (
+            None,
             None,
             f"Error: {upload_error} Please try again.",
             [],
+            {"display": "none"},
             "none",
             None,
             {"display": "none"},
         )
 
+    # Change orientation of pipeline dataframe dictionary to enable storage as JSON data
+    for key in pipelines_dict:
+        pipelines_dict[key] = pipelines_dict[key].to_dict("records")
+
     session_opts = [{"label": ses, "value": ses} for ses in sessions]
-    dataset_summary = util.construct_summary_str(data)
+    dataset_summary = util.construct_summary_str(overview_df)
 
     return (
-        data.to_dict("records"),
+        overview_df.to_dict("records"),
+        pipelines_dict,
         None,
         session_opts,
+        {"display": "block"},
         "csv",
         dataset_summary,
         {"display": "block"},
     )
 
 
+@app.callback(
+    [
+        Output("pipeline-dropdown-container", "children"),
+        Output("interactive-datatable", "style_filter_conditional"),
```

Comment:
why is this an output of a callback? Couldn't you just statically set this where the data table is defined? Unless I misunderstand, there won't be any condition where you now want the data table filter functionality enabled. So setting it statically would make more sense I think

Reply:
It's because I'm only disabling the filter functionality for columns that describe specific pipelines, e.g. `fmriprep-20.2.7` and `freesurfer-6.0.1`, aka cols that I have implemented dropdowns + backend filtering logic for. So currently there are subject identifier columns like `participant_id`, `bids_id` where users can still use the built-in filtering for if they are looking for the info of a specific subject, for example (i.e., "simple" within-column filtering).

The reason why this disabling is done dynamically as opposed to where the datatable is defined in the `layout` is just because I don't know which pipelines are available in a given dataset/datatable (and therefore which cols to disable) until after the data is uploaded + parsed.

I think one alternative in order to be able to set this statically in the layout definition would be to define the list of columns that I _don't_ want to disable, and basically make the condition something like `if not in [list_of_cols]`. The issue is that this list may need to be updated whenever I implement a new column in the overview table.

Any thoughts on this @surchs?

Reply:
oh no, I think that's a good reason. I just didn't catch that!